src/stan/mcmc/hmc/nuts/adapt_dense_e_nuts.hpp

```diff
@@ -29,12 +29,15 @@ class adapt_dense_e_nuts : public dense_e_nuts<Model, BaseRNG>,
       this->stepsize_adaptation_.learn_stepsize(this->nom_epsilon_,
                                                 s.accept_stat());
 
-      bool update = this->covar_adaptation_.learn_covariance(
-          this->z_.inv_e_metric_, this->z_.q);
+      Eigen::MatrixXd inv_metric;
+      
+      bool update = this->covar_adaptation_.learn_covariance(inv_metric, this->z_.q);
```

Comment:
Why switch to an empty matrix here?

Reply:
The inv_metric argument to the learn_covariance gets written to. It's like an output parameter.